### ISSUE TITLE
Unable to reproduce test numbers

### ISSUE THREAD

**Issue by albertfgu** [HeddaCohenIndelman/Learning-Gumbel-Sinkhorn-Permutations-w-Pytorch#1]
Thanks for your effort in porting this task to Pytorch! I was able to get the code running with the requirements specified (except I'm using Pytorch 1.1 instead of 1.0), but the numbers I'm getting when training from scratch are extremely inconsistent.
For example, with the default parameters given in the repo, I usually get a `prop_wrong` error of 0.04 and `prop_any_wrong` error of 1.0, but occasionally there will be a "good" run with no error.
Also, when varying N I have difficulty getting 0 test error; it seems very sensitive to hyperparameters like learning rate and number of epochs.
Are both of these phenomena consistent with your observations? If so, do you have general guidelines on setting the hyperparameters?

**Comment by HeddaCohenIndelman** [comment]
Hello @albertfgu,
Sorry for the late response!
The performance on the test set seems indeed to decrease as the number of random samples (n_numbers) increases. This phenomenon is also consistent with the original tensorflow implementation of the authors of the original paper (see https://github.com/google/gumbel_sinkhorn).

I analyzed the stability the algorithm from 200 repetitions of training and testing with the below settings for variable n_numbers :
training settings:
lr = 0.1, temperature = 1.0,  batch_size = 10, prob_inc = 1.0, samples_per_num = 5, 
n_iter_sinkhorn = 20, n_units =32, noise_factor= 1.0, optimizer = 'adam', keep_prob = 1.
n_epochs = 500 

testing settings:
batch_size = 1, samples_per_num = 1
 
The following summarizes the percentage of perfect test set ordering as n_numbers increases:
![GS_stability](https://user-images.githubusercontent.com/46455293/62524101-bcb4eb80-b83d-11e9-8e9d-8d1be9983944.png)
Therefore, it sorts up to 100 random samples, which is consistent with the findings of the authors.

Best, Hedda.
